src/composables/useApi.ts

```diff
@@ -0,0 +1,63 @@
+// libraries
+import { Notify } from 'quasar';
+import { api } from '../boot/axios';
+import { i18n } from '../boot/i18n';
+
+// config
+import { rideToWorkByBikeConfig } from '../boot/global_vars';
+const { apiVersion } = rideToWorkByBikeConfig;
+
+// types
+import type { Method } from 'axios';
+
+interface ApiResponse<T> {
+  data: T | null;
+}
+
+export const useApi = () => {
+  const apiFetch = async <T>({
+    endpoint,
+    payload,
+    translationKey,
+    method = 'get',
+  }: {
+    endpoint: string;
+    payload: unknown;
+    translationKey: string;
+    method: Method;
+  }): Promise<ApiResponse<T>> => {
+    try {
+      const response = await api<T>({
+        url: endpoint,
+        method: method,
+        data: payload,
+        headers: {
+          Accept: `application/json; version=${apiVersion}`,
+        },
+      });
+
+      if (response.status >= 200 && response.status < 300) {
+        Notify.create({
+          message: i18n.global.t(`${translationKey}.apiMessageSuccess`),
+          color: 'positive',
+        });
+        return { data: response.data };
+      } else {
+        Notify.create({
+          message: i18n.global.t(`${translationKey}.apiMessageError`),
+          color: 'negative',
+        });
+        return { data: null };
+      }
+    } catch (e) {
+      console.error(e);
```

Comment:
This ended up being harder than expected.
The use of logger instance depends on injecting it with Vue `inject` method, which is only available within setup function (inside a component).
Because `useApi` is called in a Pinia store, this does not work and returns a warning:
`[Vue warn]: inject() can only be used inside setup() or functional components.`
There are issues for this problem [here](https://github.com/MarcSchaetz/vuejs3-logger/issues/1) and [here](https://github.com/MarcSchaetz/vuejs3-logger/issues/2), but the solutions either don't work (such as the use of `createLogger` function which is not available in the library), or seem unusual, such as [importing straight into Pinia](https://github.com/MarcSchaetz/vuejs3-logger/issues/1#issuecomment-1516172215) within a single boot file.

Reply:
You can use Vue `inject()` func successfully:

./src/components/login/FormLogin.vue 

```vue
setup() {
    const logger = inject('vuejs3-logger');
    const loginStore = useLoginStore();
    loginStore.logger = logger;
```

./store/login.ts (see `apiFetch` func `logger` param)

```ts
const { data } = await apiFetch<LoginResponse>({
        endpoint: routesConf.api_login.path,
        method: 'post',
        payload: this.user,
        translationKey: 'login',
        logger: this.logger,
      });
```

Reply:
Ok, then I would suggest initializing the logger object in the store from a **layout** component (currently `LoginRegisterLayout`), so that we can rely on `this.logger` being available, and do not have to initialize it in each component that uses `useApi`.

Reply:
If you want all store instances has logger instance by default another solution is:

./src/boot/pinia.js

```js
import { markRaw } from 'vue';

export default boot(({ app }) => {
  const pinia = createPinia();
  // plugins
  pinia.use(piniaPluginPersistedstate);
  pinia.use(({ store }) => {
    store.$log = markRaw(app.config.globalProperties.$log);
  });
  app.use(pinia);
});
```

Logger access inside every store as:

```ts
this.$log
```

Reply:
Yes, this seems like the best way. Updated.